### PR TITLE
Add a CRO_DEBUG env var to toggle on template hot reload

### PR DESCRIPTION
**Pull request by vendethiel**

### CODE REVIEW

File: lib/Cro/WebApp/Template/Repository.pm6 [hunk 1]
```diff
@@ -42,6 +42,10 @@ monitor Cro::WebApp::Template::Repository {
         }
     }
 
+    method refresh($abs-path) {
+        %!abs-path-to-compiled{$abs-path}:delete with %!abs-path-to-compiled{$abs-path}
```
Comment (vendethiel) [inline lib/Cro/WebApp/Template/Repository.pm6:46]: removing that `with xxx` errors with "Cannot look up attributes in a Promise type object"

File: lib/Cro/WebApp/Template/Repository.pm6 [hunk 2]
```diff
@@ -52,7 +56,21 @@ monitor Cro::WebApp::Template::Repository {
...
+        if (%!abs-path-to-mtime{$abs-path} // 0) != $modified {
+            self.refresh($abs-path)
+        }
+        %!abs-path-to-mtime{$abs-path} = $modified;
+        callsame
+    }
+
+}
+
+my $template-repo = %*ENV<CRO_DEBUG> ?? Cro::WebApp::Template::ReloadingRepository.new !! Cro::WebApp::Template::Repository.new;
```
Comment (vendethiel) [inline lib/Cro/WebApp/Template/Repository.pm6:73]: maybe `CRO_DEBUG` should be stored somewhere else

Comment (jnthn) [reply lib/Cro/WebApp/Template/Repository.pm6:73]: Maybe, will have to think about that (though it's not really a blocker). I would rather it was called `CRO_DEV` though; it's a feature aimed at development time in general, not just for debugging.

File: lib/Cro/WebApp/Template/Repository.pm6 (hunk 1 above)
Comment (jnthn) [reply lib/Cro/WebApp/Template/Repository.pm6:46]: Hm, what's the stack trace (maybe with --ll-exception)? I'm surprised...

**Review by jnthn — COMMENTED** [review]
Looks good to me, just one tweak...

**Comment by vendethiel** [comment]
Test failures look spurious here too.

**Review by jnthn — APPROVED** [review]
Seems to work well, and all tests pass in local testing. In it goes. :partying_face:

**Pull request merged**